src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    [EventSource(Name = "System.Net.Http")]
+    internal sealed class HttpTelemetry : EventSource
+    {
+        public static readonly HttpTelemetry Log = new HttpTelemetry();
+
+        private IncrementingPollingCounter? _requestsPerSecondCounter;
+        private PollingCounter? _startedRequestsCounter;
+        private PollingCounter? _currentRequestsCounter;
+        private PollingCounter? _abortedRequestsCounter;
+
+        private long _startedRequests;
+        private long _currentRequests;
+        private long _abortedRequests;
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        // NOTE
+        // - The 'Start' and 'Stop' suffixes on the following event names have special meaning in EventSource. They
+        //   enable creating 'activities'.
+        //   For more information, take a look at the following blog post:
+        //   https://blogs.msdn.microsoft.com/vancem/2015/09/14/exploring-eventsource-activity-correlation-and-causation-features/
+        // - A stop event's event id must be next one after its start event.
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void RequestStart(string host, int port, string scheme, string path, string version)
```

Comment:
Is the version a major/minor pair? There is likely a small perf optimization to pass this as a pair of shorts rather than a string. The string incurs one or more heap allocations, some formatting, a couple buffer copies and a few extra bytes in the final encoding.

Reply:
Good idea. HTTP versions are typically like 1.0, 1.1, 2.0, 3.0. So, this sounds like a good optimization.